File: python/find_duplicates.py
def find_duplicates_nested_loop(l: list) -> list:
    unique = []
    duplicates = []
    for i in l:
        if (duplicates.count(i) > 0): # it has already been acounted for ignore it
            continue
        elif (unique.count(i) > 0): # it has already been accounted for once before
            duplicates.append(i)
        unique.append(i) # add it to the unique list (I FORGOT THIS INITIALLY)
    return duplicates

if __name__ == "__main__":
    sample1 = [3, 7, 5, 6, 7, 4, 8, 5, 7, 66]
    sample2 = [3, 5, 6, 4, 4, 5, 66, 6, 7, 6]
    sample3 = [3, 0, 5, 1, 0]
    sample4 = [3]
    
    print("Sample 1:", find_duplicates_nested_loop(sample1))
    print("Sample 2:", find_duplicates_nested_loop(sample2))
    print("Sample 3:", find_duplicates_nested_loop(sample3))
    print("Sample 4:", find_duplicates_nested_loop(sample4))

File: java/FindDuplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FindDuplicates {

    public static List<Integer> findModeNestedLoops(List<Integer> l) {
        List<Integer> unique = new ArrayList<Integer>();
        List<Integer> duplicates = new ArrayList<Integer>();
        for (int i = 0; i < l.size(); i++) {
            if (duplicates.contains(l.get(i))){ // check to see if its already been noted as a duplicate
                continue;
            }
            else if (unique.contains(l.get(i))) { // if it already contains that number ie its not unique and its not already been
                                                  // accounted for as a duplicate then we need to add it
                duplicates.add(l.get(i));
            }
       unique.add(l.get(i));
    }
    return duplicates; //return the numbers that appeared more than once 
    }
    public static void main(String[] args) {
        // some test strings:
        List<Integer> sample1 = new ArrayList<Integer>(Arrays.asList(3, 7, 5, 6, 7, 4, 8, 5, 7, 66));
        List<Integer> sample2 = new ArrayList<Integer>(Arrays.asList(3, 5, 6, 4, 4, 5, 66, 6, 7, 6));
        List<Integer> sample3 = new ArrayList<Integer>(Arrays.asList(3, 0, 5, 1, 0));
        List<Integer> sample4 = new ArrayList<Integer>(Arrays.asList(3));
        System.out.println("Sample 1: " + findModeNestedLoops(sample1));
        System.out.println("Sample 2: " + findModeNestedLoops(sample2));
        System.out.println("Sample 3: " + findModeNestedLoops(sample3));
        System.out.println("Sample 4: " + findModeNestedLoops(sample4));
    }

}